### DISCUSSION TITLE
Next.js suddenly does not work in WSL 2 (can't open localhost in browsers) if I turn on VPN

### DISCUSSION THREAD

**Discussion by dorklord23** [vercel/next.js#17675] [Help] [answered]
Hi, guys.

I have Next.js repo and have been working on it in WSL 2 without problem. Suddenly, today it doesn't work anymore. Next.js successfully compiles and listens to port 3000 as usual but I can't open the localhost in my browsers as if no local server is running. 

![image](https://user-images.githubusercontent.com/17370471/95301508-df5feb80-08aa-11eb-9a94-1745c585da59.png)

**Comment by lfades** [comment]
This might be an issue with the OS you're using in WSL 2, try running another program and see if it works.

I've been using WSL 2 for a long time 🤔

**Reply by dorklord23** [reply]
Hi,

Thanks for the suggestion. As it turned out, I needed to restart the computer. It took two hours of me figuring out while the solution was so simple lol.

**Comment by dorklord23** [comment, marked as answer]
Okay. After further usage, I found out that if I turn on VPN (I use Avast), browser will not be able to open localhost. Weirdly enough, it only affects port that is used by Next.js. Port that is used by create-react-app can be open in browsers.

**Reply by dorklord23** [reply]
> I have same issue. I use Astrill VPN and I can't open `localhost:3000` that was started by Next.js if I turn on the VPN. The port that is used by `create-react-app` works well with VPN. Even after restarting my PC, the same issue happens for me. Did you find the correct reason?

Sadly, I don't remember the solution/reason at all. It's been quite a while for me. I think I just turned off the VPNS then.

**Comment by msm-shubham-tamrakar-01** [comment]
Can confirm this still exists for me.

Windows 11

```
WSL version: 2.6.3.0
Distro: Ubuntu 24.04.4 LTS

Package manager: pnpm
```

Steps I tried

1. Remove node_modules, lock file and clean cache then install packages and run
2. wsl --shutdown
